### ISSUE TITLE
signal 9 connected to ProtectedDataWillBecomeUnavailable?

### ISSUE THREAD

**Issue by Simon-Laux** [deltachat/deltachat-ios#1984]
maybe we need to quit quicker on `applicationProtectedDataWillBecomeUnavailable` (happens when I lock the phone)?

```
19:29:50:967 💙  ⬅️ applicationProtectedDataWillBecomeUnavailable
19:29:52:701 💙  ⬅️ remaining background time: 12.034014750039205
19:29:55:878 💙  ⬅️ few background time, 8.867681916686706, stopping
19:29:55:885 💙  event: src/accounts.rs:271: Stopping IO for all accounts.
19:29:55:889 💙  ⬅️
19:29:55:892 💙  ⬅️ event emitter finished
19:29:59:068 💙  ⬅️ few background time, 5.666139250039123, done
Message from debugger: Terminated due to signal 9
```

`signal 9` is bad right?

**Comment by hpk42** [comment]
On Thu, Nov 16, 2023 at 10:40 -0800, Simon Laux wrote:
> maybe we need to quit quicker on `applicationProtectedDataWillBecomeUnavailable` (happens when I lock the phone)?
> 
> ```
> 19:29:50:967 💙  ⬅️ applicationProtectedDataWillBecomeUnavailable
> 19:29:52:701 💙  ⬅️ remaining background time: 12.034014750039205
> 19:29:55:878 💙  ⬅️ few background time, 8.867681916686706, stopping
> 19:29:55:885 💙  event: src/accounts.rs:271: Stopping IO for all accounts.
> 19:29:55:889 💙  ⬅️
> 19:29:55:892 💙  ⬅️ event emitter finished
> 19:29:59:068 💙  ⬅️ few background time, 5.666139250039123, done
> Message from debugger: Terminated due to signal 9
> ```
> 
> `signal 9` is bad right?

yes, it's SIGKILL, so the process got hard-killed from the OS. 
(kill -L gives at least on linux a list of signal-number/names)

**Comment by r10s** [comment]
- when logging "stopping", delta-ios has called `dc_stop_io()`
- when logging "done" 3 seconds later, delta-ios has called [`UIApplication.shared.endBackgroundTask()`](https://developer.apple.com/documentation/uikit/uiapplication/1622970-endbackgroundtask) and returns the callback signalling we're don

then there is the exit - so, it seems sth. is still running that needs to be terminated, the known issue.

i would suspect currently, that it has sth. to do with "quit quicker" or "locking" - apple does not complain all the ~20 seconds we're running but only exactly at the moment where we say we're done - but probably are not from the view of apple

@Simon-Laux but maybe you can test that the same way: see what happens if we quit already after 1 second or so after applicationProtectedDataWillBecomeUnavailable (in non-lock-situations, it is on purpose to use the 30 seconds we have after we get to background - ppl usually switch to other apps and then 2 seconds later back - this is why we're using that time)

**Comment by r10s** [comment]
[documentation](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623019-applicationprotecteddatawillbeco) says:

> If your app is currently accessing a protected file, you can use this method to release any references to that file. Although it is not an error to access the file while the device is locked, any attempts to do so will fail

also, for all our files, protection should not apply once the device get locked: https://developer.apple.com/documentation/uikit/protecting_the_user_s_privacy/encrypting_your_app_s_files# - by default, files are only protected as long as the device is the first time unlocked, they stay accessible to the all in locked and unlocked state

so, there is no hint we should do sth. special on `applicationProtectedDataWillBecomeUnavailable` currently, the kill-signal might have been raised be because of other reasons, that might have been fixed by 1.42.7. closing this issue as there is no actionable item left
